tests/test_config.py

```diff
@@ -43,3 +43,22 @@ def test_super_url_not_required(mocker):
 
     assert settings['db.common.url'] == common_url
     assert settings['db.super.url'] == common_url
+
+
+def test_with_existing_settings(mocker):
+    common_url = 'postgresql:///common'
+    other_url = 'oracle:///common'
+
+    _patch = {
+        'DB_URL': other_url,
+        'DB_SUPER_URL': other_url,
+    }
+    mocker.patch.dict(os.environ, _patch)
+    existing_settings = {
+        'db.common.url': common_url
+    }
+
+    settings = discover_settings(existing_settings)
+
+    assert settings['db.common.url'] == common_url
```

Comment:
To me, it's more natural if the environment variable `DB_SUPER_URL` overrides the existing settings... :thinking:

Reply:
The use case for this is that the application may want to get its setting elsewhere, say from a config file (e.g. config.ini under db-connection-string). The application needs to be able to provide an overridden setting without paving over the user defined environment variable.